ext/ddtrace.c

```diff
@@ -1367,6 +1367,108 @@ static void dd_disable_if_incompatible_sapi_detected(void) {
     }
 }
 
+static php_stream_wrapper ddtrace_original_http_wrapper;
+static php_stream_wrapper ddtrace_original_https_wrapper;
+static php_stream_wrapper ddtrace_traced_http_wrapper;
+static php_stream_wrapper ddtrace_traced_https_wrapper;
+
+static php_stream *ddtrace_stream_opener(
+    php_stream_wrapper *wrapper,
+    const char *filename,
+    const char *mode,
+    int options,
+    zend_string **opened_path,
+    php_stream_context *context
+) {
+    if (!context) {
+        context = php_stream_context_alloc();
+    }
+
+    // Inject distributed tracing headers
+    zval *http_context_zv = zend_hash_str_find(Z_ARRVAL(context->options), "http", sizeof("http") - 1);
+    if (!http_context_zv) {
+        zval tmp;
+        array_init(&tmp);
+        zend_hash_str_update(Z_ARRVAL(context->options), "http", sizeof("http") - 1, &tmp);
+        http_context_zv = zend_hash_str_find(Z_ARRVAL(context->options), "http", sizeof("http") - 1);
+    }
+
+    if (http_context_zv && Z_TYPE_P(http_context_zv) == IS_ARRAY) {
+        HashTable *http_context = Z_ARRVAL_P(http_context_zv);
+        ddtrace_inject_distributed_headers(http_context, HEADER_MODE_CONTEXT);
+    }
+
+    // Open internal span
+    ddtrace_span_data *span = ddtrace_open_span(DDTRACE_INTERNAL_SPAN);
```

Comment:
You should actually use `ddtrace_alloc_execute_data_span` and ddtrace_clear_execute_data_span() here, so that manual instrumentation (or instrumentation from appsec tracing these things) shares a span with this.

Use index -2.

Reply:
Also, note that you then cannot use `_add_new`, because it already might exist on the span.